src/shared_state.rs

```diff
@@ -111,13 +123,30 @@ impl Server {
                 connect_op.pass = Some(pass);
             }
             AuthStyle::Token(token) => connect_op.auth_token = Some(token),
-            _ => {}
-        }
+            AuthStyle::Credentials(path) => {
+                let contents = fs::read_to_string(&path)?;
+                let re = Regex::new(CREDS_FILE_REGEX).unwrap();
+                let captures = re.captures_iter(&contents).collect::<Vec<_>>();
+
+                let (user_jwt, seed) = match &captures[..] {
+                    [jwt, seed, ..] => (jwt[1].to_string(), seed[1].to_string()),
+                    _ => {
+                        return Err(Error::new(
+                            ErrorKind::InvalidInput,
+                            format!("cannot parse credentials in {}", path.display()),
+                        ))
+                    }
+                };
 
-        let op = format!(
-            "CONNECT {}\r\nPING\r\n",
-            serde_json::to_string(&connect_op)?
-        );
+                let key_pair = KeyPair::from_seed(&seed)
+                    .map_err(|err| Error::new(ErrorKind::InvalidData, err))?;
+                nkey = Some(key_pair);
+
+                jwt = Some(user_jwt);
```

Comment:
Where do we sign the nonce?

Reply:
What we want ideally is that this is loaded on demand only when needed to form CONNECT {}.